server/server/controllers/customer.js

```diff
@@ -186,12 +186,52 @@ async function deleteCart(req, res) {
   }
 }
 
+async function checkout(req, res) {
+  const customerId = req.user.userId;
+  try {
+    //get the customer and fill the basket with the items data
+    const customer = await customerModel
+      .findById(customerId)
+      .populate("basket.items.itemId");
+    const basketItems = customer.basket.items;
+    if (basketItems.length < 1) {
+      return res.status(301).json({ message: "basket is empty!" });
+    }
+    //create the order object and fill it with data from the basket
+    const order = {};
```

Comment:
Required restaurantId from the items in the basket that will be ordered

Reply:
added and pushed